counter/src/main/java/org/infinispan/counter/impl/listener/CounterManagerNotificationManager.java

```diff
@@ -69,15 +67,15 @@ public CounterManagerNotificationManager() {
    }
 
    /**
-    * The executor to use where the user's {@link CounterListener} is invoked.
+    * The blockingManager to use where the user's {@link CounterListener} is invoked.
     *
-    * @param asyncExecutor The {@link Executor} implementation.
+    * @param blockingManager The {@link BlockingManager} to use.
     */
-   public void useExecutor(Executor asyncExecutor) {
-      if (asyncExecutor == null) {
+   public void useBlockingManager(BlockingManager blockingManager) {
+      if (blockingManager == null) {
          return;
       }
-      userListenerExecutor = new LimitedExecutor("counter-listener", asyncExecutor, 1);
+      userListenerExecutor = blockingManager.limitedBlockingExecutor("counter-listener", 1);
```

Comment:
I didn't notice this before, are we assuming that the listeners are blocking?

Reply:
Yeah, the counter listener hasn't been updated to support non blocking.
I have created https://issues.redhat.com/browse/ISPN-11720